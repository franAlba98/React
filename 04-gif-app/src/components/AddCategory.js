import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ anyNameICanUseTheNameOfTheUseState }) => {

    const [inputValue, setInputValue] = useState('Hello world');

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        //preventDefault() prevents the browser from refreshing each time something is submitted in the form
        e.preventDefault();

        if (inputValue.trim().length > 2) {

            //The thing here is that I call the prop anyNameICan... that I pass as an argument in the component (line 3). The problem is that I have no access to the array of cities in GifExpertApp
            //component, I could also pass the array as an argument but is not required because with referencing the setCategories using the anyNameICan with the callback "categories". This gives
            //me access to the information of the previous state.
            anyNameICanUseTheNameOfTheUseState(categories => [...categories, inputValue]);
            setInputValue('');

        }


    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )

}

AddCategory.propTypes = {
    anyNameICanUseTheNameOfTheUseState: PropTypes.func.isRequired
}
