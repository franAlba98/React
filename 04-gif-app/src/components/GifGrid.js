import React, { useEffect, useState } from 'react'

export const GifGrid = ({ category }) => {

    //a conter button for tests sake
    const [count, setcount] = useState(0);


    //The use effect is used so when firing an event in the component (for example pressing a button) the getGifs won't be executed again. This way
    //we tell react that we want to make this request just one time
    useEffect(() => {
        getGifs();
        //The secon argument here (the empty array) specifies the useEffect dependencies. When using an empty array we tell react that useEffect doesn't 
        //have any dependencies so the getGifs must be executed a single time.
    }, []);

    const getGifs = async () => {

        const url = "https://api.giphy.com/v1/gifs/search?q=bongo+cat&limit=10&api_key=a7md0mTdcGuIQIFv0Bi1KbZYvgGANprI";
        const res = await fetch(url);
        const { data } = await res.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);


    }

    //getGifs();


    return (
        <div>
            <h3>{category}</h3>
            <h3>{count}</h3>
            <button onClick={() => setcount(count + 1)}></button>
        </div>
    )
}
