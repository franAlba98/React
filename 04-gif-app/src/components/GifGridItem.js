import React from 'react'

export const GifGridItem = ({ id, title, url }) => {

    //console.log(id,title,url);

    return (
        //next line the content of <div></div> uses the css class 'box' to make it look nicer
        <div className='box'>
            <img src={url} alt={title} />
            <p> {title} </p>
        </div>
    )
}
