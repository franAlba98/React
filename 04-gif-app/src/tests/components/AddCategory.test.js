import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"


describe('AddCategory component tests', () => {

    //the function I need to pass to AddCategory
    const anyNameICanUseTheNameOfTheUseState = () => {

    }

    test('must show correctly', () => {
        const wrapper = shallow(<AddCategory anyNameICanUseTheNameOfTheUseState={anyNameICanUseTheNameOfTheUseState} />);

        expect(wrapper).toMatchSnapshot();

    })


})
