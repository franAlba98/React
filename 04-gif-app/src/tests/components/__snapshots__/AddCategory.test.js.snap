// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component tests must show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
